pom.xml

```diff
@@ -944,7 +945,7 @@
                                     <version>${maven.version.range}</version>
                                 </requireMavenVersion>
                                 <requireJavaVersion>
-                                    <version>${java.version}</version>
+                                    <version>${compile.java.version}</version>
```

Comment:
Hi @zhfeng 
I prefer to set `8` as `requireJavaVersion`. Java 11 is required only when compiling parser modules.
Sometimes we just need to compile a part of modules (such as protocol, agent), build an image, do some tests, which could be done using Java 8. Adding `-Denforcer.skip=true` is not elegant.
And the error of compiling antlr 4.10+ with Java 8 is clear enough for developers.
We could clarify Java requirement in [Wiki](https://github.com/apache/shardingsphere/wiki) and [docs](https://shardingsphere.apache.org/community/en/involved/contribute/dev-env/).

Reply:
Or is it possible to add `maven-enforcer-plugin` with Java 11 requirement to those modules require Java 11? The other modules keep using Java 8

Reply:
Yeah, I think it makes sense and I will investigate.